crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -425,6 +424,34 @@ pub struct SuiObject<T: SuiMoveObject> {
     pub reference: SuiObjectRef,
 }
 
+impl TryInto<Object> for SuiObject<SuiRawData> {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Object, Self::Error> {
+        let data = match self.data {
+            SuiRawData::MoveObject(o) => {
+                let struct_tag = parse_struct_tag(o.type_())?;
+                Data::Move(unsafe {
+                    MoveObject::new_from_execution(
+                        struct_tag,
+                        o.has_public_transfer,
+                        o.version,
+                        o.child_count,
+                        o.bcs_bytes,
+                    )
+                })
+            }
```

Comment:
What is this used for? The point of new_from_execution is to ensure that the metadata is valid and controlled/validated by the VM. For example, `has_public_transfer` is uniquely determined by the `struct_tag` and very bad things will happen if those are out of match.

Reply:
in the RPC server, we convert `Object` to `SuiObject` for better json serialisation. This code here does the reverse, converting `SuiObject` back to `Object` on the Rust client, this is needed to be able to reuse move's lib's `MoveStruct::simple_deserialize` and `MovePackage::disassemble` on the client side.